lib/libfrr.c

```diff
@@ -902,11 +902,14 @@ static int frr_config_read_in(struct thread *t)
 	 * reading the configuration file.
 	 */
 	if (frr_get_cli_mode() == FRR_CLI_TRANSACTIONAL) {
+		struct nb_context context = {};
+		char errmsg[BUFSIZ] = {0};
```

Comment:
why can't we have errmsg part of context structure.

Reply:
That would simplify things a bit but I don't think the error message belongs in what we're calling the "northbound context". I think it's cleaner to have the error message as a separate output parameter. It would be ugly, for example, to have the northbound callbacks logging errors to `args->context->errmsg` instead of `args->errmsg`.